model/product.ts

```diff
@@ -12,91 +12,166 @@ type Product = {
     product_image: string;
     stock: number;
 };
+
+type Review = {
+    review_id: number;
+    author: string;
+    rating: number;
+    comment: string;
+};
+
+type ProductWithReviews = Product & {
+    reviews: Review[];
+};
+
```

Comment:
Will a product without reviews still show up in this? Checking as it seems all of the queries now use the ProductsWithReviews type

Reply:
Good question. 

It will return the product with review fields as null:

![image](https://github.com/fac29/hoda-e-commerce-backend/assets/97399953/e3c3d763-53f2-4b80-8937-d715dac71fc8)

I'm guessing we can go ahead with that?